ui/app/services/websockets.js

```diff
@@ -0,0 +1,54 @@
+import Service from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { prettyPrintJson } from 'pretty-print-json';
+import { htmlSafe } from '@ember/template';
+import { run, schedule } from '@ember/runloop';
+
+export default class WebsocketsService extends Service {
+  _socket = null;
+
+  @tracked
+  isTrackingRecords = true;
+
+  @tracked
+  _records = [];
+
+  connect(url) {
+    if (this._socket) {
+      return this._socket;
+    } else {
+      this._socket = new WebSocket(url);
+    }
+
+    this._socket.addEventListener('message', (event) => {
+      if (!this.isTrackingRecords) {
+        return;
+      }
+
+      const data = JSON.parse(event.data);
+
+      run(() => {
+        if (this._records.length >= 10) {
```

Comment:
Just curious: are records with a length of less than 10 characters usually just meta data or something else that's not really needed in the stream view?

Reply:
Its not the length of the record, but the length of the records array. We only hold in memory and display the last 10 records from the websocket

Reply:
Got it and thank you for the outline!